### PR TITLE
Optimize the way GlanceAPISpec is built

### PR DESCRIPTION
**Pull request by fmount**
As done for `Cinder`, this patch reduces the need to repeat the same fields many times in different structures, especially in the `GlanceAPISpec`. In particular, `GlanceAPISpec` includes a `GlanceAPITemplate` which represents a list of parameters defined for the specific instance, while other parameters (e.g. `ExtraMounts` or `PasswordSelectors`) are inherited from the top level `CR`.

### CODE REVIEW

File: controllers/glance_controller.go [hunk 1]
```diff
@@ -653,6 +643,13 @@ func (r *GlanceReconciler) apiDeploymentCreateOrUpdate(instance *glancev1.Glance
 		// and what is inherited from the top-level CR (ExtraMounts)
 		deployment.Spec = apiSpec
 
+		// We might want to create instances pointing to different backends in
+		// the future, hence we inherit the customServiceConfig (where the backends
+		// are defined) only if it's not specified in the GlanceAPITemplate
+		if len(deployment.Spec.CustomServiceConfig) == 0 {
```
[the remark this answers is not in the document]
Comment (konan-abhi) [reply controllers/glance_controller.go:649]: Just one question (may be not related to this PR), if we have configured HA x3 then will this service config will be common for all 3 glance instances or we can differentiate the service config for each instance?

[the remark this answers is not in the document]
Comment (fmount) [reply controllers/glance_controller.go:649]: That's exactly the point: you can customize the config for each instance, but you can choose to not do that and "propagate" the same config to the underlying instances (regardless of the `Replicas` or the HA model you choose)

[the remark this answers is not in the document]
Comment (konan-abhi) [reply controllers/glance_controller.go:649]: Makes sense, thank you!

**Review by konan-abhi — APPROVED** [review]
Looks good to me, thank you for explaining in detail about custom configs.

**Pull request merged**